lib/tasks/eu_exit_business_finder.rake

```diff
@@ -21,9 +21,40 @@ namespace :eu_exit_business_finder do
       updated_links = updated_links - EuExitFacetMigrationConfig::facet_values_to_remove
       updated_links = updated_links.uniq
       list.write_attribute('links', facet_values: { any: updated_links })
+      list.write_attribute('title', title_from_facets(updated_links))
       list.save
     end
   end
+
+  # Title is constructed from finder content item, so we have to retrieve the
+  # facet titles from the JSON using their content IDs.
+  def title_from_facets(chosen_facet_ids)
+    business_finder_json = Services.publishing_api.get_content('42ce66de-04f3-4192-bf31-8394538e0734')
+    all_facets = business_finder_json['links']['facet_group'][0]['links']['facets']
+    all_facet_ids_and_titles = flatten_facet_values_to_h(all_facets)
+    chosen_facet_ids_and_titles = override_special_titles(all_facet_ids_and_titles.filter { |content_id, _title|
+      chosen_facet_ids.include? content_id
+    })
+    chosen_facet_titles = chosen_facet_ids_and_titles.map { |_content_id, title| title }
+    comma_separated_facet_titles = chosen_facet_titles.map { |choice_name| "'#{choice_name}'" }.join(", ")
+    "EU Exit guidance for your business in the following #{'category'.pluralize(chosen_facet_titles.count)}: #{comma_separated_facet_titles}"
+  end
+
+  def flatten_facet_values_to_h(facets)
+    array_of_facet_value_hashes = facets.map do |facet|
+      facet['links']['facet_values'].map { |facet_value| [facet_value['content_id'], facet_value['title']] }.to_h
```

Comment:
You could override the special values here, which means the `override_special_titles` method could be totally removed.
```
facet['links']['facet_values'].map { |facet_value| [facet_value['content_id'], facet_value_labels_to_replace.fetch(facet_value['content_id'], facet_value['title'])] }.to_h
```